TYPO3.Flow/Classes/TYPO3/Flow/Object/Proxy/ProxyClass.php

```diff
@@ -206,7 +206,12 @@ public function render()
         $namespace = $this->namespace;
         $proxyClassName = $this->originalClassName;
         $originalClassName = $this->originalClassName . \TYPO3\Flow\Object\Proxy\Compiler::ORIGINAL_CLASSNAME_SUFFIX;
-        $abstractKeyword = $this->reflectionService->isClassAbstract($this->fullOriginalClassName) ? 'abstract ' : '';
+        $classModifier = '';
+        if ($this->reflectionService->isClassAbstract($this->fullOriginalClassName)) {
+            $classModifier = 'abstract ';
+        } elseif ($this->reflectionService->isClassFinal($this->fullOriginalClassName)) {
+            $classModifier = 'final ';
+        }
```

Comment:
This moves and `final` modifier from the `*_Original` class to the proxy, just like we already did with the `abstract` modifier.